### PR TITLE
MKLDNN BatchNorm operator refactor

### PR DESCRIPTION
**Pull request by grygielski**
This PR is next in series of refactoring, templatizing and preparing code for easier MKLDNN v1.0 update.

It provides refactor of Batch Normalization MKLDNN operator.

### CODE REVIEW

**Review by jczaja — APPROVED** [review]
LGTM

**Comment by jczaja** [comment]
@luotao1 Could you please review this PR?
